faunadb-java-dsl/src/main/java/com/faunadb/client/types/Properties.java

```diff
@@ -0,0 +1,203 @@
+package com.faunadb.client.types;
+
+import com.faunadb.client.errors.FaunaException;
+import com.faunadb.client.types.Types.SimpleType;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.*;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Maps.newHashMap;
+
+class Properties {
```

Comment:
Seems like this could be private and final. Also with a private default constructor.